### PR TITLE
feat: google and facebook login setup

### PR DESCRIPTION
**Pull request by Enubia**
- [x] add google/fb login buttons
- [x] create routing for both

~~handle logout~~ no logout needed since it would also logout the user from every other open session

~~we still need a facebook test account and need to enable the local environments in google dashboard so that we can proceed with the development @stacysealky~~ this is done

### CODE REVIEW

File: nodemon.json [hunk 1]
```diff
@@ -3,6 +3,6 @@
     "server",
     "client"
   ],
-  "ext": "js, ejs",
+  "ext": "js",
```
Comment (Enubia) [inline nodemon.json:6]: @gnorbsl no need to restart the server on ejs file changes

Comment (gnorbsl) [reply nodemon.json:6]: Ah makes sense, I somehow thought that pages get rendered once on server start :D

**Comment by Enubia** [comment]
@gnorbsl @stacysealky please review

File: nodemon.json (hunk 1 above)
Comment (Enubia) [reply nodemon.json:6]: nope, only on load in browser, we don't use webpack yet for hot reloading

**Comment by stacysealky** [comment]
@Enubia This looks great. Reviewed 16/20 files. @gnorbsl can you review the 4 files that aren't reviewed yet?

File: server/db/repository/UserRepository.js [hunk 2]
```diff
@@ -3,31 +3,31 @@ const User = require('../models/User');
 
 async function getUserByObjectId(id) {
   try {
-    return User.findOne({ _id: id });
+    return User.findOne({ _id: id }).exec();
```
Comment (gnorbsl) [inline server/db/repository/UserRepository.js:6]: why the exec now?

Comment (Enubia) [reply server/db/repository/UserRepository.js:6]: https://mongoosejs.com/docs/promises.html#queries-are-not-promises

**Pull request merged**